synfig-core/src/synfig/color/colorblendingfunctions.h

```diff
@@ -37,87 +37,85 @@
 
 namespace synfig {
 
-typedef Color (*blendfunc)(Color &,Color &,float);
+typedef Color (*blendfunc)(Color, Color, float);
 
 template <class C>
-C blendfunc_COMPOSITE(C &src,C &dest,float amount)
+C blendfunc_fg(C fg, C bg, float amount)
 {
-	//c_dest'=c_src+(1.0-a_src)*c_dest
-	//a_dest'=a_src+(1.0-a_src)*a_dest
-
-	float a_src=src.get_a()*amount;
-	float a_dest=dest.get_a();
-	const float one(C::ceil); 
+	return fg;
+}
 
-	// if a_arc==0.0
-	//if(fabsf(a_src)<COLOR_EPSILON) return dest;
+template <class C>
+C blendfunc_bg(C fg, C bg, float amount)
+{
+	return bg;
+}
 
-	// Scale the source and destination by their alpha values
-	src*=a_src;
-	dest*=a_dest;
+template <class C>
+C blendfunc_COMPOSITE(C fg, C bg, float amount)
+{
+	const float a_fg = fg.get_a() * amount;
+	const float a_bg = bg.get_a();
 
-	dest=src + dest*(one-a_src);
+	const float one(C::ceil); 
 
-	a_dest=a_src + a_dest*(one-a_src);
+	const float a = a_fg + a_bg * (one - a_fg);
 
-	// if a_dest!=0.0
-	if(fabsf(a_dest)>COLOR_EPSILON)
+	if(fabsf(a) > COLOR_EPSILON)
 	{
```

Comment:
prefer this style:

```c++
if (fabsf(a) > COLOR_EPSILON) {
```

(space before parenthesis and curly bracket in the same line)

Reply:
Okay, will keep in mind and make the change.

Reply:
Any updates?

Reply:
Sorry, I was working on the opengl thing and forgot about this change. Will make it today and push.